message_sender/tasks.py

```diff
@@ -147,18 +142,23 @@ def decr_message_count(cls, msg_type, msg_time):
             return
         bucket = int(msg_time // cls.BUCKET_SIZE)
 
-        key = cls.get_key(msg_type, bucket)
+        key = cls.get_key(channel.channel_id, bucket)
         # Set the expiry time to the timeout minus the time passed since
         # the message was sent.
         if int(cache.get_or_set(key, 0, timeout - time_since)) > 0:
             cache.decr(key)
 
     @classmethod
-    def manage_limit(cls, task, msg_type, limit, timeout, delay):
+    def manage_limit(cls, task, channel):
+        limit = channel.concurrency_limit
+        timeout = channel.message_timeout
+        delay = channel.message_delay
+
         if limit > 0:
-            if cls.get_current_message_count(msg_type, timeout) >= limit:
+            if cls.get_current_message_count(
+                    channel.channel_id, timeout) >= limit:
                 task.retry(countdown=delay)
-            cls.incr_message_count(msg_type, timeout)
+            cls.incr_message_count(channel.channel_id, timeout)
```

Comment:
since the timeout is specified on the channel why not pass `channel` here rather than the id and the timeout? (like you've done for `decr_message_count`)